services/src/main/java/org/fao/geonet/api/records/MetadataWorkflowApi.java

```diff
@@ -419,23 +442,268 @@ public MetadataStatus convertParameter(int id, MetadataStatusParameter parameter
+            // If metadata record does not exists then it was deleted so
+            // we will only allow the administrator, owner to view the contents
+            checkCanViewStatus(fullMetadataText, metadataStatus, httpSession, request);
+        }
+
+        return fullMetadataText;
+    }
+
+    @ApiOperation(
+            value = "Get saved content from the status record after changes",
+            notes = "",
+            nickname = "showStatusAfter")
+    @RequestMapping(value = "/{metadataUuid}/status/{statusId:[0-9]+}.{userId:[0-9]+}.{changeDate}/after",
+            method = RequestMethod.GET,
+            produces = {
+                    MediaType.APPLICATION_XML_VALUE
+            })
+
+    @PreAuthorize("hasRole('Editor')")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Version of the record after changes."),
+            @ApiResponse(code = 403, message = ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_EDIT)
+    })
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public String showStatusAfter(
+            @ApiParam(value = API_PARAM_RECORD_UUID, required = true) @PathVariable String metadataUuid,
+            @ApiParam(value = "Status identifier", required = true) @PathVariable int statusId,
+            @ApiParam(value = "User identifier", required = true) @PathVariable int userId,
+            @ApiParam(value = "Change date", required = true) @PathVariable String changeDate,
+            @ApiIgnore @ApiParam(hidden = true) HttpSession httpSession, HttpServletRequest request
+    )
+            throws Exception {
+        MetadataStatus metadataStatus;
+
+        metadataStatus = metadataStatusRepository
+                .findOneByUuidAndStatusValue_IdAndUserIdAndChangeDate(metadataUuid, statusId, userId, new ISODate(changeDate));
+
+        if (metadataStatus == null) {
+            throw new ResourceNotFoundException(
+                    String.format("Can't find metadata status for record '%d', user '%s' at date '%s'. Changes cannot be viewed", metadataUuid,
+                            userId, changeDate));
+        }
+
+        if (!StatusValueType.event.equals(metadataStatus.getStatusValue().getType()) || !ArrayUtils.contains(supportedRestoreStatuses, statusId)) {
+            throw new NotAllowedException("Unsupported action on status type '" + metadataStatus.getStatusValue().getType() + "' for metadata '" + metadataStatus.getUuid() + "'. Supports status type '" +
+                    StatusValueType.event + "' with the status id '" + Arrays.toString(supportedRestoreStatuses) + "'.");
+        }
+
+        if (metadataStatus.getCurrentState() == null) {
+            throw new ResourceNotFoundException(
+                    String.format("No data exists for metadata record '%d', user '%s' at date '%s'. Changes state cannot be viewed", metadataUuid,
+                            userId, changeDate));
+        }
+
+        String fullMetadataText = metadataStatus.getCurrentState();
```

Comment:
To check to refactor with the code https://github.com/geonetwork/core-geonetwork/pull/4817/files#diff-668fdf0a09c1a8afb5771c9e85715598R496-R506, seem the same code

Reply:
refactored code for before/after api so that that they use common code.